File: Bubble Sort + Selection Sort + Insertion Sort/js/InsertionSort.js

let arr = [7 ,8, 3, 1, 2];

for(let i=1; i<arr.length; i++) {

    let current = arr[i];
    let j = i-1;

    while(j>=0 && arr[j] > current) {
        arr[j + 1] = arr[j];
        j--;
    }

    // correct position pe current ko daalo
    arr[j+1] = current;
}

console.log("Sorted Array using Insertion Sort : ", arr)

File: Bubble Sort + Selection Sort + Insertion Sort/py/InsertionSort.py

arr = [7, 8, 3, 1, 2] 
n = len(arr) 


# insertio sort logic 
for i in range(1, n): # start from second element 
    key = arr[i] # current element to be inserted 
    j = i - 1
    
    # move elements that are greater than key to one position ahead 
    
    while j>=0 and arr[j] > key:
        arr[j+1] = arr[j]
        j -= 1
        
    arr[j+1] = key # insert key at correct position 
    

print("Sorted Array : ", arr)